src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/OpenSslPkcs12Reader.cs

```diff
@@ -132,5 +124,56 @@ public List<OpenSslX509CertificateReader> ReadCertificates()
 
             return certs;
         }
+
+        private static bool TryRead(byte[] data, out OpenSslPkcs12Reader pkcs12Reader, out Exception openSslException, bool captureException)
+        {
+            SafePkcs12Handle handle = Interop.Crypto.DecodePkcs12(data, data.Length);
+            openSslException = null;                
+
+            if (!handle.IsInvalid)
+            {
+                pkcs12Reader = new OpenSslPkcs12Reader(handle);
+                return true;
+            }
+
+            handle.Dispose();
+            pkcs12Reader = null;
+            if (captureException)
+            {
+                openSslException = Interop.Crypto.CreateOpenSslCryptographicException();
+            }
+            else
+            {
+                Interop.Crypto.ErrClearError();
+            }
+
+            return false;
+        }
+
+        private static bool TryRead(SafeBioHandle fileBio, out OpenSslPkcs12Reader pkcs12Reader, out Exception openSslException, bool captureException)
+        {
+            SafePkcs12Handle p12 = Interop.Crypto.DecodePkcs12FromBio(fileBio);
+            openSslException = null;
+
+            if (!p12.IsInvalid)
+            {
+                pkcs12Reader = new OpenSslPkcs12Reader(p12);
+                openSslException = null;                
```

Comment:
This setting to null isn't needed, or alternatively moved the earlier setting to null down to the else block below (in which case it'd be good to keep the previous method that follows a similar form in sync with this pattern).